src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -91,7 +120,7 @@ internal override ConstantValue ExplicitDefaultConstantValue
                 // Dev11 emits the first parameter as option with default value and the second as regular parameter.
                 // The syntactic default value is suppressed since additional synthesized parameters are added at the end of the signature.
 
-                return DefaultSyntaxValue ?? DefaultValueFromAttributes;
+                return OptionalParameter.ConstantValue ?? DefaultValueFromAttributes;
```

Comment:
Considering this task, and the issue of default parameters in abstract methods, you said that an additional pass may be necessary. As a status update, I've pushed a change that addresses the concerns about caching `BoundExpression`s (disables caching entirely) - however, I'd like to discuss how to properly address the issue of another pass: if it's necessary, how to implement it, etc.